src/bin.ts

```diff
@@ -73,13 +73,17 @@ export function bootstrap(state: BootstrapState) {
   if (!state.phase2Result) {
     state.phase2Result = phase2(state);
     if (state.shouldUseChildProcess && !state.isInChildProcess) {
-      return callInChild(state);
+      // Note: When transitioning into the child-process after `phase2`,
+      // the updated working directory needs to be preserved.
+      return callInChild(state, state.phase2Result.targetCwd);
```

Comment:
This explicit cwd is strictly not needed since spawn would just use the working directory we switched to, but I think being explicit is helping with readability and also makes this safer